Ginger/Ginger/RunSetPageLib/NewRunSetPage.xaml.cs

```diff
@@ -442,77 +443,77 @@ private void UpdateRunButtonIcon(bool isRunStarted = false)
-        //                xReRunFailedRunsetBtn.ButtonImageForground = (SolidColorBrush)FindResource("$SelectionColor_LightBlue");
-        //                xReRunFailedRunsetBtn.IsEnabled = false;
-        //                if (RunSetConfig.GingerRunners.Any(x => x.Executor.IsRunning == true))
-        //                {
-        //                    xStopRunsetBtn.ButtonText = "Stop";
-        //                    xStopRunsetBtn.ButtonImageType = eImageType.Stop;
-        //                    xStopRunsetBtn.ButtonStyle = (Style)FindResource("$RoundTextAndImageButtonStyle_ExecutionStop");
-        //                    xStopRunsetBtn.IsEnabled = true;
-        //                    xStopRunsetBtn.Visibility = Visibility.Visible;
-        //                }
-        //                xContinueRunsetBtn.Visibility = Visibility.Collapsed;
-        //                xResetRunsetBtn.Visibility = Visibility.Collapsed;
-        //            }
-        //            else
-        //            {
-        //                xReRunFailedRunsetBtn.ButtonImageType = eImageType.Run;
-        //                xReRunFailedRunsetBtn.ButtonStyle = (Style)FindResource("$RoundTextAndImageButtonStyle_Execution");
-        //                xReRunFailedRunsetBtn.ButtonImageForground = (SolidColorBrush)FindResource("$SelectionColor_Pink");
-        //                xReRunFailedRunsetBtn.IsEnabled = true;
-        //                xStopRunsetBtn.Visibility = Visibility.Collapsed;
-        //                xContinueRunsetBtn.Visibility = Visibility.Visible;
-        //                xResetRunsetBtn.Visibility = Visibility.Visible;
-        //                xRunsetSaveBtn.IsEnabled = true;
-        //            }
-        //        }
-        //        else 
-        //        { 
-        //            xReRunFailedRunsetBtn.Visibility = Visibility.Collapsed;
-        //            xRunRunsetBtn.Visibility = Visibility.Visible;
-        //        }
-        //    });
-        //}
+        private void UpdateReRunFailedButtonIcon(bool isRunStarted = false)
+        {
+
+            this.Dispatcher.Invoke(() =>
+            {
+                bool setAsRunning = false;
+                if (WorkSpace.Instance.RunsetExecutor.Runners.Any(x => x.Executor.BusinessFlows.Any(x => x.RunStatus == eRunStatus.Failed)))
+                {
+                    xReRunFailedRunsetBtn.Visibility = Visibility.Visible;
+                    xRunRunsetBtn.Visibility = Visibility.Collapsed;
+
+                    if (isRunStarted)
+                    {
+                        xReRunFailedRunsetBtn.ButtonText = "Starting...";
+                        xReRunFailedRunsetBtn.ToolTip = "Performing Run Preparations";
+                        setAsRunning = true;
+                    }
+                    else if (mRunSetConfig.IsRunning)
+                    {
```

Comment:
this and below else if block looks similar, please merge these conditions if there is no significant difference

Reply:
Done